### ISSUE TITLE
Loading overlay does not get removed when data added via transaction

### ISSUE THREAD

**Issue by bpfoster** [ag-grid/ag-grid#8358]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Requesting Community Support**

If you are not a customer of AG Grid, ag-grid staff will label your issue as managed-by-the-community. This means that AG Grid staff is not going to be actively looking into it and it will get closed if inactive for more than one month. The community is welcome to help with this question/support issue.

**Providing a Reproducible Scenario**
Accepted reproducible scenarios are
- I am upgrading from ag-grid 31 to 32
- We apply all data, even the initial set, with `applyTransaction()`
- If you start with `null` `rowData` and call `applyTransaction({ add: items })`, the grid populates but the loading overlay stays present

I took the overlay plunker and modified it slightly to provide `null` initial rowData and to apply data via transaction:
https://plnkr.co/edit/4WoVVfXFe6vYLr7v?open=main.js

**Current behavior**
Loading overlay stays present unless you explicitly call `gridApi.setGridOption('loading', false);`

**Expected behavior**
As in v31, the loading overlay gets removed after data is loaded via transaction.

**Please tell us about your environment:**

* **AG Grid version:** X.X.X
32.0.2

* **Browser:**
Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by AG-Zoheil** [comment]
Hi @bpfoster,

Thank you for raising this on GitHub.

This is the expected behaviour.
If you want to show an overlay specifically for no rows/data please use the "No rows overlay". If the loading overlay is set to true it will display on top of any rows.

Kind regards,
Zoheil

**Comment by pingjuh** [comment]
Hi @AG-Zoheil , I'm facing this issue too but I can't find AG-12327 in https://www.ag-grid.com/pipeline/. Has this been fixed? Thanks

**Comment by Dimitri-Habruk** [comment]
> y to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next relea

Hi, any updates?

**Comment by AG-Zoheil** [comment]
Hi,

This was fixed in version `32.1.0`.

Thanks!
